### PR TITLE
Export environment variables in a cross-compatible way

### PR DESCRIPTION
**Pull request by Mogztter**
@rahmanusta Could you please tell me if this is working for you (ie. Windows) ?

### CODE REVIEW

**Comment by rahmanusta** [comment]
Hey @Mogztter. It works very well. Thanks for this work!

**Comment by Mogztter** [comment]
You're welcome, this is not fully working yet, I will try to fix it this week

**Comment by Mogztter** [comment]
This is harder than I thought, I will accept your pull request

**Comment by mojavelinux** [comment]
Amazing that Node makes this so hard. Seems like something that should be a core feature.

**Comment by Mogztter** [comment]
Indeed, I tried several approaches, but none of them worked. `cross-env` package seems to be the "least worst solution" :wink:

File: npm/arg.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    // Filter out this argument from process.argv
+    return false;
+  }
+  return true;
+}
+
+// Export environement variables from command line arguments.
+// Argument starting with 'env.' will be removed from process.argv and added to process.env
+Arg.prototype.exportEnvironmentVariables = function() {
+  var args = process.argv.slice(2);
```
Comment (mojavelinux) [inline npm/arg.js]: Why are we using slice here instead of considering every argument?

**Comment by mojavelinux** [comment]
I had one question about the implementation. Also, I kind of prefer a better name than Args. Perhaps we can name the module something more explanatory?

File: npm/arg.js (hunk 1 above)
Comment (Mogztter) [inline npm/arg.js]: The first argument is `node` and the second is the program name. `process.argv.slice(2)` gives me the remaining arguments in the command line. Eg:

```javascript
process.argv = ["node", "test.js", "env.MINIFY=1", "env.PRODUCTION=1" ];
process.argv.slice(2) // [ "env.MINIFY=1", "env.PRODUCTION=1" ]
```

**Comment by Mogztter** [comment]
@mojavelinux I've canceled this pull-request and merged https://github.com/asciidoctor/asciidoctor.js/commit/9fdb21c5315718b7facc867229e32e4c23beab24 instead.

**Comment by mojavelinux** [comment]
Oh, my mistake! Thanks for the clarifications!

**Pull request closed without merging**